### PR TITLE
fix: remove kubelet flag azure-container-registry-config

### PR DESCRIPTION
**Pull request by AlisonB319**
**What type of PR is this?**
We are seeing test failures due to
```
Dec 14 23:23:35 9yh7-2024-12-14-ubuntu2204gen2containerdairgappedk8snotca000000 kubelet[14328]: E1214 23:23:35.541222   14328 run.go:74] "command failed" err="failed to parse kubelet flag: unknown flag: --azure-container-registry-config"
Dec 14 23:23:35 9yh7-2024-12-14-ubuntu2204gen2containerdairgappedk8snotca000000 systemd[1]: kubelet.service: Main process exited, code=exited, status=1/FAILURE
```

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: e2e/scenario_test.go [hunk 1]
```diff
@@ -633,6 +633,9 @@ func Test_Ubuntu2204Gen2_ContainerdAirgappedK8sNotCached(t *testing.T) {
 			Cluster: ClusterKubenetAirgap,
 			VHD:     config.VHDUbuntu2204Gen2ContainerdAirgappedK8sNotCached,
 			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				if nbc.KubeletConfig != nil {
+					delete(nbc.KubeletConfig, "--azure-container-registry-config")
```
Comment (djsly) [inline e2e/scenario_test.go:637]: this flag was removed in AKS 1.31, shouldnt we use that condition to decide when not to configure it.

File: e2e/node_config.go [hunk 2]
```diff
@@ -381,7 +381,6 @@ func baseTemplateLinux(location string, k8sVersion string) *datamodel.NodeBootst
 			"--anonymous-auth":                    "false",
 			"--authentication-token-webhook":      "true",
 			"--authorization-mode":                "Webhook",
-			"--azure-container-registry-config":   "/etc/kubernetes/azure.json",
```
Comment (cameronmeissner) [inline e2e/node_config.go:384]: should we make this conditional on the k8s version (1.30 should be the pivot I think), and also do the same for windows? (in this same file)

Comment (AlisonB319) [reply e2e/node_config.go:384]: Where are we actually setting the k8s version?

File: e2e/node_config.go [hunk 3]
```diff
@@ -448,13 +449,16 @@ func baseTemplateLinux(location string, k8sVersion string) *datamodel.NodeBootst
...
+	config, err := pruneKubeletConfig(k8sVersion, config)
+	require.NoError(t, err)
+	return config
 }
 
 // this been crafted with a lot of trial and pain, some values are not needed, but it takes a lot of time to figure out which ones.
 // and we hope to move on to a different config, so I don't want to invest any more time in this-
-func baseTemplateWindows(location string) *datamodel.NodeBootstrappingConfiguration {
+func baseTemplateWindows(t *testing.T, location string) *datamodel.NodeBootstrappingConfiguration {
 	kubernetesVersion := "1.29.9"
```
Comment (cameronmeissner) [inline e2e/node_config.go:460]: we should probably have this version also be resolved dynamically from the e2e cluster, out of scope for this PR though

File: e2e/node_config.go [hunk 4]
```diff
@@ -760,3 +767,20 @@ func zipWindowsCSE() (*os.File, error) {
 
 	return zipFile, nil
 }
+
+// https://dev.azure.com/msazure/CloudNativeCompute/_git/aks-rp?path=/resourceprovider/server/microsoft.com/datamodel/model/defaults_kubelet.go&version=GBmaster&_a=contents
```
Comment (cameronmeissner) [inline e2e/node_config.go:771]: let's avoid internal links since this OSS :)

**Review by djsly — APPROVED** [review]

**Review by cameronmeissner — APPROVED** [review]
might consider making the constraint `>= 1.30`

**Review by djsly — APPROVED** [review]

**Pull request merged**